### ISSUE TITLE
Consecutive `POST /comments` get 200 instead of 201 in a period of time.

### ISSUE THREAD

**Issue by southp** [WP-API/WP-API#1660]
## Environment
WordPress: 4.3.1
WP REST API: 2.0-beta 4
Machine: [Chassis](https://github.com/Chassis/Chassis) with VirtualBox 4.3.28 as its VM backend

## How To
Here is the snippet of the shell script I used to experiment. It uses [httpie](httpie.org). Simply replace the url and the post id should do the trick:

```
#!/bin/sh
date_str=`date`
http POST http://<site-url>/wp-json/wp/v2/comments author_name='Shell' \
    author_email='shell@xd.cc' \
    author_url='http://goog.le' \
    content="from shell $date_str" \
    post=<pid> \
    parent=0
```
In my case:

1. Run the script the first time. It should return `201`, and the anonymous comment is created successfully.
1. Run the script again. It returned 200 without response body.
1. Wait for a period of time ( about 30sec ~ 1 min. ). Now you should return to the step 1.

**Comment by Toddses** [comment]
This is probably default WordPress behavior. Anonymous comments in WordPress are limited to one every X seconds. I think the default is 15 seconds.

**Comment by danielbachhuber** [comment]
Core calls `wp_die()` for this check ([ref](https://core.trac.wordpress.org/browser/tags/4.3/src/wp-includes/comment.php#L1501)), so we'll need to address it when we decide how to handle `wp_die()` #790

For now, I'll add a skipped test #1662

**Comment by rmccue** [comment]
Per https://github.com/WP-API/WP-API/issues/790#issuecomment-73872576 can we avoid this function ever getting called? We should be able to run the `comment_flood_filter` filter ourselves before `wp_allow_comment` (which eventually triggers the action) is run.

**Comment by danielbachhuber** [comment]
> can we avoid this function ever getting called?

Avoid calling `check_comment_flood_db()` entirely? I should think we need to persist that behavior.

**Comment by websupporter** [comment]
We could utilize `wp_allow_comment()` in `create_item_permissions_check()`, if the risk of running into `wp_die()` wouldn't be there. Since 4.4 we have `wp_handle_comment_submission()` which handles the exit process in `wp-comments-post.php`. So, if we could transform `wp_die()` for flood and duplicate comments into an WP Error object, which travels all the way to `wp-comments-post.php`, this would be the step to go, right?

I've tried to get this started here: https://core.trac.wordpress.org/ticket/36901

**Comment by rachelbaker** [comment]
Fixed in #2795
